[EDU-2244] BUILD MENU PT Rules Engine (#335)
aziontech/docs: src/content/docs/pt-br/pages/build-journey/edit-edge-app/edit-rules-engine/rules-engine.mdx

---

> title: Como criar regras para automatizar comportamentos usando o Rules Engine

Reviewer: ```suggestion
title: Como criar regras para automatizar comportamentos com o Rules Engine
```

description: Integre regras em sua aplicação para definir tarefas para cenários específicos sem alterar seu código existente.
meta_tags: build
namespace: docs_guides_build_work_with_rules_engine

> permalink: /documentation/products/guides/build/work-with-rules-engine/

Author: ```suggestion
permalink: /documentacao/produtos/guias/build/trabalhe-com-rules-engine/
```

> permalink: /documentation/products/guides/build/work-with-rules-engine/

Reviewer: ajustar link

menu_namespace: buildMenu
---

> Implementar regras em sua aplicação usando o [Rules Engine](/en/documentation/products/edge-application/rules-engine/) permite que você determine as tarefas que ela deve executar em cenários específicos sem modificar o código-fonte da sua aplicação.

Reviewer: ```suggestion
Implementar regras em sua aplicação usando o [Rules Engine](/pt-br/documentacao/produtos/edge-application/rules-engine/) permite que você determine as tarefas que ela deve executar em cenários específicos sem modificar o código-fonte da sua aplicação.
```

:::caution[warning]

> Para desbloquear todas as funcionalidades do Rules Engine, você deve ativar o módulo [Application Acceleration](/en/documentation/products/edge-application/application-acceleration/). Se o **Application Acceleration** estiver ativado, a transferência de dados pode gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/en/pricing/) para mais informações.

Reviewer: ```suggestion
Para desbloquear todas as funcionalidades do Rules Engine, você deve ativar o módulo [Application Acceleration](/pt-br/documentacao/produtos/edge-application/application-acceleration/). Se o **Application Acceleration** estiver ativado, a transferência de dados pode gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/pt-br/precos/) para mais informações.
```

:::

## Via RTM

Este guia demonstra o processo de criação de regras com um exemplo prático, mas a maioria das regras será única para cada aplicação e suas necessidades. Primeiro, para ativar o módulo **Application Acceleration**:

1. [Acesse o RTM](https://manager.azion.com).
2. No canto superior esquerdo da página, abra o **Products menu**, representado por três linhas horizontais, e selecione **Edge Application**.
3. Clique na edge application que você deseja configurar.
4. Ative o módulo **Application Acceleration**.
5. Clique no botão **Save**.

> Além de ativar variáveis como [device groups](/en/documentation/products/guides/build/create-device-groups/), [configurações de cache](/en/documentation/products/edge-application/cache-settings/) e [origens](/en/documentation/products/guides/build/work-with-origins/), você pode criar uma grande variedade de regras nas fases de solicitação e resposta.

Reviewer: lembrar ajustar links PT

Este guia ajudará você a criar uma regra na fase de resposta para remover um cabeçalho HTTP usando o comportamento **Filter Response Header**. O cabeçalho `Server` armazena informações sobre o servidor que gerou a resposta. Ao remover este cabeçalho, os usuários finais não receberão informações sobre a infraestrutura da sua aplicação, aumentando sua segurança ao reduzir as informações disponíveis para possíveis atacantes.

:::note
Cuidado ao remover qualquer cabeçalho de sua edge application para que esta ação não interfira em nenhum padrão e diretriz aplicável.
:::

1. Navegue até a aba **Rules Engine**.
2. Clique no botão **Add Rule**.
3. No menu suspenso, selecione **Response Phase**.
4. Dê um nome à sua regra e, se necessário, uma descrição.
5. Na seção **Criteria**, selecione a variável `${uri}`.
6. Como operador de comparação, selecione **começa com**.
7. Como argumento, adicione o valor `/`.
8. Na seção **Behaviors**, selecione **Filter Response Header**.
9. Como argumento, adicione `Server`.
10. Clique no botão **Save**.

## Via API

> 1. Execute a seguinte solicitação `PATCH` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/en/documentation/products/guides/personal-tokens/) e a variável `<edge_application_id>` por [seu ID de edge application(/en/documentation/products/guides/build/configure-main-settings/) para ativar o módulo [Application Acceleration](/en/documentation/products/edge-application/application-acceleration/):

Reviewer: ```suggestion
1. Execute a seguinte solicitação `PATCH` em seu terminal, substituindo `[TOKEN VALUE]` por seu [personal token](/pt-br/documentacao/produtos/gestao-de-contas/personal-tokens/) e a variável `<edge_application_id>` por [seu ID de edge application(/en/documentation/products/guides/build/configure-main-settings/) para ativar o módulo [Application Acceleration](/pt-br/documentacao/produtos/edge-application/application-acceleration/):
```

--header 'Content-Type: application/json' \
--data '{
    "name": "FilterRespHeader Server",
    "behaviors": [
        {
            "name": "filter_response_header",
            "target": "Server"
        }
    ],
    "criteria": [
        [
            {
                "variable": "${uri}",
                "operator": "is_equal",
                "conditional": "if",
                "input_value": "/"
            }
        ]
    ]
}'
```

4. Você receberá a seguinte resposta:

```json
{
    "results": {
        "id": <rule_id>,
        "name": "FilterRespHeader Server",
        "phase": "response",
        "behaviors": [
            {
                "name": "filter_response_header",
                "target": "Server"
            }
        ],
        "criteria": [
            [
                {
                    "variable": "${uri}",
                    "operator": "is_equal",
                    "conditional": "if",
                    "input_value": "/"
                }
            ]
        ],
        "is_active": true,
        "order": 1,
        "description": null
    }
}
```

5. Aguarde alguns minutos para que as alterações se propaguem.

:::tip

> Consulte a [documentação da API Azion](https://api.azion.com/) e a [especificação OpenAPI](https://github.com/aziontech/azionapi-openapi/) para saber mais sobre todos os recursos disponíveis via API.

Reviewer: ```suggestion
Consulte a [documentação da API da Azion](https://api.azion.com/) e a [especificação OpenAPI](https://github.com/aziontech/azionapi-openapi/) para saber mais sobre todos os recursos disponíveis via API.
```
